### ISSUE TITLE
Error 500 during posting and listing 

### ISSUE THREAD

**Issue by psolovie** [NuGet/NuGetGallery#7536]
NuGet.org returns error 500 during posting and listing via api and nuget.org UI. Error is spontaneous and usually can be fixed by a retry of the action.

This behavior breaks continues delivery automation.

![image](https://user-images.githubusercontent.com/55137163/64606079-1aed6500-d3ce-11e9-96ec-12305096387f.png)

**Comment by loic-sharma** [comment]
Thank you for reporting this, we will investigate this issue.

**Comment by loic-sharma** [comment]
Do you have a package that reproduces this issue? I believe this is because we're returning a HTTP Status string that's [longer than 512 characters](https://docs.microsoft.com/en-us/dotnet/api/system.web.httpresponse.statusdescription?redirectedfrom=MSDN&view=netframework-4.8#exceptions).

**Comment by psolovie** [comment]
Hello, stable reproduce with listing for package https://www.nuget.org/packages/intelmkl.devel.linux-x64/2019.5.281/Manage

**Comment by psolovie** [comment]
When I make api call to list the package I get an error 'An unlisted package should never be latest or latest stable'

**Comment by psolovie** [comment]
Error is related to block of code 

if (!package.Listed && (package.IsLatestStable || package.IsLatest || package.IsLatestSemVer2 || package.IsLatestStableSemVer2))
            {
                throw new InvalidOperationException("An unlisted package should never be latest or latest stable!");
            }

in https://github.com/NuGet/NuGetGallery/blob/e3ae678fd8bc9f1065a17ccfb0ab3be82328bf55/src/NuGetGallery.Services/PackageManagement/PackageUpdateService.cs

I have an idea what's going on: for some reason after package version  https://www.nuget.org/packages/intelmkl.devel.linux-x64/2019.5.281/Manage was unlisted, current version for package https://www.nuget.org/packages/intelmkl.devel.linux-x64 is still 2019.5.281

so we have:

1. Unlist operation for some reason brake data consistency for package - does not update latest version property - that's the root cause of the bug
2. api returns a valid error
3. UI crashes because of message length limitations

I ask you to fix data issue ASAP to allow me to mark package version  https://www.nuget.org/packages/intelmkl.devel.linux-x64/2019.5.281/Manage as listed. That blocks project deliverable

**Comment by loic-sharma** [comment]
Your package was indeed in a bad state. I went ahead and fixed that up, you should be able to list your package now. Please let me know if you run into any additional issues!

**Comment by psolovie** [comment]
We have 2 more packages with the same issue. Please fix ASAP.
https://www.nuget.org/packages/inteldaal.devel.osx-x64/
https://www.nuget.org/packages/inteldaal.redist.win-x64/

**Comment by loic-sharma** [comment]
I reflowed the packages you linked. These packages should be fixed if they were affected by the same issue as before. /cc @scottbommarito who will be on-call later today.

**Comment by scottbommarito** [comment]
@psolovie - We are unsure of what bug is causing the data issue, but to mitigate any issues associated with it, we are removing the error that you are encountering. As a result, if your packages get into this situation again, you will be able to list/unlist your package without issues. It will, however, take some time before this additional change is deployed.

To investigate how the data is getting into this invalid state, it would be helpful if you could give us instructions on how to repro the issue. It seems very likely to us that this bug is associated with performing many actions simultaneously--for example, if you are pushing and unlisting packages simultaneously, you may be exposing a concurrency bug with the code that determines which package is latest.

**Comment by psolovie** [comment]
Scenario: 

1. packages (up to 20) are simultaneously uploaded to NuGet with api: PUT https://www.nuget.org/api/v2/package 
2. immediately after step 1 uploaded packages are simultaneously unlisted with api: DELETE https://www.nuget.org/api/v2/package/"packageId"
3. after some time (a couple of days) packages are listed with api: POST https://www.nuget.org/api/v2/package/"packageId" 

Step 3 returns 500 for some packages
Also in some cases step 2 returns 404 error, so we have to await for some time before start package unlisting

**Comment by scottbommarito** [comment]
@psolovie - so you're calling PUT and DELETE almost simultaneously? That's likely causing our issue--the push is still ongoing when the delete is called. This is the same reason why you're getting a 404 sometimes as well. We apologize for the issue, but unfortunately, bugs like these are very difficult to fix. Could you wait a short period of time (maybe a couple seconds) between pushing and unlisting each package? You shouldn't encounter this bug anymore once you do that.

**Comment by psolovie** [comment]
Thank you for the information. I have added time intervals between PUT and DELETE. Will have results after the next uploading session.

**Comment by scottbommarito** [comment]
The endpoint should no longer throw. Unfortunately I can't verify the behavior as I don't own any packages that are in this weird state. I'm going to go ahead and close this issue, because the 500 and the exception that were blocking you have been fixed, but I have created another issue for the purpose of tracking the weird "IsLatest" behavior when uploading and unlisting simultaneously: https://github.com/NuGet/NuGetGallery/issues/7607. Please provide further updates to that issue.
